index.js

```diff
@@ -12,6 +14,19 @@ var diagnostics = require("./lib/diagnostics");
 var logs = require("./lib/logs");
 var extensions = require("./lib/extensions");
 var webjobs = require("./lib/webjobs");
+var Promise = require("bluebird");
+
+function promisifyApi(target) {
+    Object.keys(target).forEach(function (key) {
+        Promise.promisifyAll(target[key]);
+    });
+
+    Object.keys(target.extensions).forEach(function (key) {
+        Promise.promisifyAll(target.extensions[key]);
+    });
```

Comment:
I thought more our discussion about documenting how to promisify the API without actually implementing it. With nuances like this (the extensions module) I do think promise support out of the box is a significant value add.